### DISCUSSION TITLE
Temperature Offset per sensor

### DISCUSSION THREAD

**Discussion by nonix** [arendst/Tasmota#12267] [Q&A] [answered]
Hi guys, I am not sure if I haven't seen this feature either proposed or already present, however would it be possible to add temp offset per sensor?

Justification: There are devices with multiple sensors attached DS18, DHT, BME, etc. sometime a combination of them. Each of the sensor could different compensation. It would be handy to have ability to set something like a calibration values per sensor rather than offsetting all of them at the same level.

What do you think?

**Comment by Jason2866** [comment, marked as answer]
Temp offset is in general a ugly/bad solution. If a sensor is "off" it is in most cases not a linear offset.
Best solution replace the sensor. A sensor with a offset out of spec. is just defect and should be replaced.
You cant really heal defect hardware with software.

**Reply by sfromis** [reply]
Indeed. On the practical Tasmota level, expanding the various offsets (several value types) for many sensors could take up a lot of extra space in the size-limited settings structure. Features eating up "too much" space here are not exactly met with open arms.

Probably having several sensors on the same board is not very common, and while I do have a couple of such boards, I just have no expectation of the sensors to precisely align. Any wild outliers would be en route to the garbage bin.

**Comment by nonix** [comment]
guys, thank you for the answer, however, there is case where you could benefit.
in a case you have an enclosure to which you have attached an "outside" sensor. The sensor reading will get affected by esp radiated heat inside. Even if the insulation and enclosure designs do reflect thermal separation. Here you could have a chip temp reading using DS18 while outside temp sensor is e.g. BME.

Anyway, I agree with code complexity and space limitations, so I will use rules as workaround.
Thx a lot,
 N.

**Reply by sfromis** [reply]
If you have a sensor mounted too close to the esp radiator, you have a bad hardware design which can not be fixed by a simple offset compensation. The error you introduced is not linear. You'd probably not get a reliable compensation using rules either.

**Comment by lalo-uy** [comment]
You may add the offsets by rules.
